lisp/core/install-file.el

```diff
@@ -0,0 +1,55 @@
+;;; core/install-file.el --- Install packages from files, .tar files, or directories  -*- lexical-binding: t; -*-
+
+;;; Commentary:
+;;
+;; Command use to install packages from files, .tar files, or directories
+;;
+;;   $ eask install-file [files..]
+;;
+;;
+;;  Positionals:
+;;
+;;    [files..]     files to install as packages; it will install through the
+;;                  function `package-install-file'
+;;
+
+;;; Code:
+
+(let ((dir (file-name-directory (nth 1 (member "-scriptload" command-line-args)))))
+  (load (expand-file-name "_prepare.el"
+                          (locate-dominating-file dir "_prepare.el"))
+        nil t))
+
+(defun eask-install-file--guess-name (file)
+  "Guess the package name of the install FILE."
+  (file-name-sans-extension (file-name-nondirectory (directory-file-name file))))
+
+(defun eask-install-file--packages (files)
+  "The file install packages with FILES."
+  (let* ((deps (mapcar (lambda (file)
+                         (list (eask-install-file--guess-name file) file))
```

Comment:
This breaks if the package name does not match the folder name, and for `tar.gz` files.

Reply:
> This breaks if the package name does not match the folder name, and for tar.gz files.

This sounds reasonable to me. Any suggestions? :)

Reply:
Yeah, I've put a fix up in #318 (copied the idea from #316). Specifically, I've made sure that the target package name must be found by `package.el` otherwise fail.